### PR TITLE
Use explicit kwarg for plt.show()

### PR DESCRIPTION
**Pull request by dstansby**
I feel like this might be a Bad Idea, but can't work out why - maybe others will be able to tell me if it is or not! Either way, the doc change is probably welcome.

Will add API notes if/when someone reviews.

### CODE REVIEW

**Comment by ImportanceOfBeingErnest** [comment]
I do think it's a good idea. Maybe one could even check for `block` being a boolean. You won't believe how many people try to do things like `plt.show(fig1)`, `plt.show(ax)` etc. Showing a useful error in such cases may be helpful for them.

File: lib/matplotlib/pyplot.py [hunk 1]
```diff
@@ -238,12 +239,14 @@ def show(*args, **kw):
...
-    described above.
+    Parameters
+    ----------
+    block : bool, optional
+        Experimental; set to ``False`` to override the blocking
+        behavior described above.
     """
     global _show
-    return _show(*args, **kw)
+    return _show(block=True)
```
[the remark this answers is not in the document]
Comment (dstansby) [reply lib/matplotlib/pyplot.py]: 🤦‍♂️

**Comment by ImportanceOfBeingErnest** [comment]
Apparently there is a problem with the web backend which doesn't take any arguments.

File: lib/matplotlib/backends/backend_webagg.py [hunk 2]
```diff
@@ -319,7 +319,10 @@ def trigger_manager_draw(manager):
         manager.canvas.draw_idle()
 
     @staticmethod
-    def show():
+    def show(block=True):
```
Comment (tacaswell) [inline lib/matplotlib/backends/backend_webagg.py]: I thought we fixed this already....

**Comment by tacaswell** [comment]
I am :-1: on changing the default value.   If the user passed `block=True` we should block unconditionally on any other state.

I am very :+1: on @ImportanceOfBeingErnest 's suggestion of giving a good warning in the common incorrect use of passing figures or axes to `plt.show` (which _does_ make sense in a guessing-the-API mode).

**Comment by dstansby** [comment]
Is there currently a difference between `block=None` and `block=True`? I thought they were the same, but maybe not.

**Comment by ImportanceOfBeingErnest** [comment]
Concerning `block=True` in the webbackend: #11201

**Comment by dstansby** [comment]
Going to close this since I can't remember what's going on.

**Pull request closed without merging**